src/input/password-input.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+
+interface Props {
+    id: string;
+    value: any;
+    onBlur?: (event: React.FormEvent<HTMLInputElement>) => void;
+    append?: string;
+    prepend?: string;
+    onChange: (value: any) => void;
+    disabled?: boolean;
+    className?: string;
+    placeholder?: string;
+    onMouseEnter?: (event: React.MouseEvent<HTMLDivElement>) => void;
+    onMouseLeave?: (event: React.MouseEvent<HTMLDivElement>) => void;
+    autoFocus?: boolean;
+    passwordVisible?: boolean;
+    passwordToggleVisible?: boolean;
+    onPasswordToggle?: (event: React.MouseEvent<HTMLDivElement>) => void;
```

Comment:
Should that come from the outside? Looks to me like that should be handled by this component internally by managing its own state in regards of showing the password or not.

Reply:
Possibly yes. I could do the management of the `type` attribute internally but still call the provided callback whenever just in case it's useful for some future feature.

Reply:
Same argument as line 16. cantradiction between naming (password) and purpose (change type). would stick with password thing.

Reply:
Man the component is called `PasswordInput` :D Anyways I switched to a stateful component. So i consider it `fixed`